File: Ex54MaterialTest/app/src/main/java/com/jmnl2020/ex54materialtest/IntroActivity.java
package com.jmnl2020.ex54materialtest;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;

import java.util.zip.Inflater;

public class IntroActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_intro);

        //4초 후에 메인액티비티 자동실행
        handler.sendEmptyMessageDelayed(0, 4000);
    }

    Handler handler = new Handler(){
        @Override
        public void handleMessage(@NonNull Message msg) {

            Intent intent = new Intent(IntroActivity.this, MainActivity.class);
            startActivity(intent);

            finish();
        }
    };

}

File: Ex91KoptlinRecyclerViewApp/app/src/main/java/com/jmnl2020/ex91koptlinrecyclerviewapp/IntroActivity.kt
package com.jmnl2020.ex91koptlinrecyclerviewapp

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_intro.*
//id 를 변수로 만들겠다!

class IntroActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_intro)

        //버튼의 id가 btn.
        // 그 버튼에 클릭 리스너 추가하기
        btn.setOnClickListener(object:View.OnClickListener{
            override fun onClick(v: View?) {
                //Main Activity 실행
                var intent= Intent(this@IntroActivity, MainActivity::class.java)
                startActivity(intent)
            }
        })
        //btn. 을 하고 거기에 onClickListener를 함 - 오브젝트를 만들겠다 object:
        //

    }

    fun clickExit(view: View) {

        finish()
        //액티비티 종료

    }
}
